low_level_platform/impl/src/lf_windows_support.c

```diff
@@ -278,10 +299,20 @@ int _lf_cond_timedwait(lf_cond_t* cond, instant_t wakeup_time) {
   }
 
   // convert ns to ms and round up to closest full integer
-  DWORD wait_duration_ms = (wait_duration + 999999LL) / 1000000LL;
+  interval_t wait_duration_ms = (wait_duration + 999999LL) / 1000000LL;
+  DWORD wait_duration_saturated;
+  if (wait_duration_ms > 0xFFFFFFFFLL) {
```

Comment:
Does this mean that we have a maximum sleep duration for windows = 2e32ms = 49 weeks (if I calculated correctly)?

Reply:
Yes.  Also, it means that on Windows, you can't get better 1ms time resolution in real time, but I guess that's OK. It's not an RTOS.